client/src/components/FileInput.vue

```diff
@@ -38,7 +38,7 @@ export default {
       let fileName = file.name.split('.')
       let extension = fileName[fileName.length - 1]
       for(let type in this.types){
-        if(this.types[type].includes(extension)){
+        if(this.types[type].extension.includes(extension)){
           return type
```

Comment:
I think this whole section could possibly be simplified using JavaScript's `some()` and `endswidth()` methods. Something like:

```javascript
if (this.types[type].extension.some(ext => extension.endswith(ext))) {
  return type;
}
```

but it's not urgent to make this change.